ww

```
@@ -211,20 +222,26 @@ if [[ -n "$IS_RUNNING" || -n "$FILTERALT" ]]; then
 	# Uses `awk` separately in order to avoid masking a return value, as Shellcheck recommends
 	ID=$(awk '{print $2}' <<< "$INFO_DBUS_SEND") || exit 1
 
-	# Try legacy first (kde <= 5) and then new (kde >= 6)
-	if dbus-send --session --dest=org.kde.KWin --print-reply=literal "/$ID" org.kde.kwin.Scripting.run 2>/dev/null; then
-    	SCRIPT_PATH="/$ID"
+	# Use kwin version to decide how to call the script run api which changes between kwin versions.
+	# See https://github.com/academo/ww-run-raise/issues/15#issuecomment-2632214974 for more info.
+	kwinVersion=$(kwin --version | cut -d ' ' -f2)
```

something odd in my arch installation, it seems that `kwin` is not an available command

```
❯ ./ww -f firefox
./ww: line 227: kwin: command not found
```

probably because I am running xorg?

```
kwin_x11 --version
``` this is available. 


I wonder if it is going to be much faster and easier to simply invoke all possible variations of the script and just let them fail silently

---

Very unfortunate, that `kwin` is not the same in every distro :/

>I wonder if it is going to be much faster and easier to simply invoke all possible variations of the script and just let them fail silently

Either that or use the kwin version detection from `INFO_ACTIVE` which seems to work for me too.

I changed the kwin version detection and fixed a bug in my PR that did not call Scripting.stop in the end.